### PR TITLE
 PLF-8383 : share news

### PR DESCRIPTION
**Pull request by azaoui**

### CODE REVIEW

File: integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 1]
```diff
@@ -164,6 +175,51 @@ public Response updateNews(@Context HttpServletRequest request,
...
   }
 
+  @POST
+  @Path("{id}/share")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "share a news", httpMethod = "POST", response = Response.class, notes = "This share a news to another space if the authenticated user is a member of the space or a spaces super manager.")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "News shared"),
+          @ApiResponse(code = 400, message = "Invalid query input"),
+          @ApiResponse(code = 401, message = "User not authorized to update the news"),
```
Comment (thomasdelhomenie) [inline integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: Missing line for error 404.

File: integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 2]
```diff
@@ -164,6 +175,51 @@ public Response updateNews(@Context HttpServletRequest request,
...
+                            @ApiParam(value = "News id", required = true) @PathParam("id") String id,
+                            @ApiParam(value = "Space list", required = true) SharedNewsEntity sharedNewsList) {
+
+    try {
+      News news = newsService.getNews(id);
+      if (news == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+
+      String authenticatedUser = ConversationState.getCurrent().getIdentity().getUserId();
```
Comment (thomasdelhomenie) [inline integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: We should use `request.getRemoteUser()` instead, it makes tests easier (no use of static methods).

File: integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 3]
```diff
@@ -164,6 +175,51 @@ public Response updateNews(@Context HttpServletRequest request,
...
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "share a news", httpMethod = "POST", response = Response.class, notes = "This share a news to another space if the authenticated user is a member of the space or a spaces super manager.")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "News shared"),
+          @ApiResponse(code = 400, message = "Invalid query input"),
+          @ApiResponse(code = 401, message = "User not authorized to update the news"),
+          @ApiResponse(code = 500, message = "Internal server error") })
+  public Response shareNews(@Context HttpServletRequest request,
+                            @ApiParam(value = "News id", required = true) @PathParam("id") String id,
+                            @ApiParam(value = "Space list", required = true) SharedNewsEntity sharedNewsList) {
```
Comment (thomasdelhomenie) [inline integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: It contains only 1 entity, so shouldn't it be called `sharedNews` ?

**Comment by azaoui** [comment]
PR moved to the branch feature/news

**Pull request closed without merging**